src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1579,15 +1608,19 @@ def create_oracledb_connection():
                 data_fetching_thread_pool_executor.shutdown(wait=True)
 
         logger.debug("All data has been successfully loaded into the Snowflake table.")
-        self._session._conn._telemetry_client.send_data_source_perf_telemetry(
-            DATA_SOURCE_DBAPI_SIGNATURE, time.perf_counter() - start_time
-        )
+        telemetry_json_string["end_to_end_duration"] = time.perf_counter() - start_time
+
         # Knowingly generating AST for `session.read.dbapi` calls as simply `session.read.table` calls
         # with the new name for the temporary table into which the external db data was ingressed.
         # Leaving this functionality as client-side only means capturing an AST specifically for
         # this API in a new entity is not valuable from a server-side execution or AST perspective.
         res_df = partitioner.driver.to_result_snowpark_df(
             self, snowflake_table_name, struct_schema, _emit_ast=_emit_ast
         )
+        telemetry_json_string["schema"] = res_df.schema.simple_string()
+        self._session._conn._telemetry_client.send_data_source_perf_telemetry(
+            telemetry_json_string
+        )
+        logger.debug(f"recorded telemetry: {json.dumps(telemetry_json_string)}")
```

Comment:
we should log something more meaningful to user instead of the whole telemetry object, like:
```python
logger.debug("local fetch started at **")
logger.debug("local fetch ended at **, total time **")
logger.debug("upload started at **")
logger.debug("upload ended at **, total time **")
logger.debug("ingestion started at **")
logger.debug("ingestion ended at, total ingestion time: **")
```